compiler/src/iree/compiler/Codegen/Dialect/GPU/TargetUtils/ConfigUtils.cpp

```diff
@@ -212,4 +214,247 @@ LogicalResult setMatmulLoweringConfig(IREE::GPU::TargetAttr target,
+                                           Operation *op) {
+  auto linalgOp = dyn_cast<linalg::LinalgOp>(op);
+  // Bail out on multi result cases as consumer fusion currently does not
+  // support multi result ops.
+  if (!linalgOp || linalgOp.getNumDpsInits() != 1) {
+    return failure();
+  }
+
+  // This pipeline requires tensor semantics. Also fail for gather semantics
+  // for now to simplify tile + fuse.
+  if (!linalgOp.hasPureTensorSemantics() || linalgOp.hasIndexSemantics()) {
+    return failure();
+  }
+
+  SmallVector<unsigned int> partitionableLoops;
+  linalgOp.getParallelDims(partitionableLoops);
+
+  // Bail out if op is not tilable.
+  if (partitionableLoops.empty()) {
+    return failure();
+  }
+
+  const int subgroupSize = target.getPreferredSubgroupSize();
+  const unsigned loopDepth = linalgOp.getNumLoops();
+
+  // Configurations we need to decide.
+  std::array<int64_t, 3> workgroupSize;
+  SmallVector<int64_t> workgroupTileSizes;
+  SmallVector<int64_t> threadTileSizes;
+
+  // Initialize the configuration.
+  auto initConfiguration = [&]() {
+    workgroupSize = {subgroupSize, 1, 1};
+    workgroupTileSizes.resize(loopDepth, 0);
+    threadTileSizes.resize(loopDepth, 0);
+
+    // Initialize tiling along all partitioned loops with size 1.
+    for (int64_t loopIndex : partitionableLoops) {
+      workgroupTileSizes[loopIndex] = threadTileSizes[loopIndex] = 1;
+    }
+    // Override the innermost dimension to distribute to threads in a subgroup.
+    workgroupTileSizes[partitionableLoops.back()] = subgroupSize;
+  };
+
+  // Common case for all linalg ops.
+
+  // The core idea is to distribute the partitioned loops to the workgroup
+  // dimensions. The goal is to fill up the GPU as much as possible, which means
+  // 1) distributing to as many threads as possible, and 2) avoid assigning too
+  // many threads to handle out-of-bound elements (thus idle).
+
+  auto elementHasPowerOfTwoBitwidth = [](Value operand) {
+    Type elementType = getElementTypeOrSelf(operand.getType());
+    return isa<IntegerType, FloatType>(elementType) &&
+           llvm::isPowerOf2_64(IREE::Util::getTypeBitWidth(elementType));
+  };
+
+  // Whether we can try to use the vectorization pipeline.
+  SmallVector<int64_t> loopBounds = linalgOp.getStaticLoopRanges();
+  bool vectorizable =
+      // Require all affine maps to be projected permutation so that we can
+      // generate vector transfer ops.
+      llvm::all_of(
+          linalgOp.getIndexingMapsArray(),
+          [](AffineMap map) { return map.isProjectedPermutation(); }) &&
+      llvm::all_of(linalgOp->getOperands(), elementHasPowerOfTwoBitwidth) &&
+      llvm::none_of(loopBounds, ShapedType::isDynamic);
+
+  const unsigned minBitwidth = getMinElementBitwidth(linalgOp);
+  // Make sure we use a tile size that results in some integral number of bytes.
+  const unsigned scaleToByte = minBitwidth < 8 ? 8 / minBitwidth : 1;
```

Comment:
```suggestion
  const unsigned scaleToByte = (minBitwidth < 8) ? (8 / minBitwidth) : 1;
```

or `max(8 / minbitwidth, 1)`?